google/resource_compute_disk.go

```diff
@@ -279,12 +287,14 @@ func resourceComputeDiskRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	var disk *compute.Disk
-	if zone, ok := d.GetOk("zone"); ok {
+	if zone, err := getZone(d, config); zone != "" && err == nil {
 		disk, err = config.clientCompute.Disks.Get(
-			project, zone.(string), d.Id()).Do()
+			project, zone, d.Id()).Do()
 		if err != nil {
 			return handleNotFoundError(err, d, fmt.Sprintf("Disk %q", d.Get("name").(string)))
 		}
+	} else if err != nil {
```

Comment:
You should remove this else if block. If no zone are specified, it is fine and the logic in the else block tries to guess the zone.

This comment also applies to group manager and autoscaler.

Reply:
I've done this as well, sorry.